gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineTesting.java

```diff
@@ -39,38 +43,60 @@ public void apply(Project project) {
             }
         });
 
-        project.getPlugins().withType(JavaPlugin.class, p -> {
-
+        project.getPlugins().withType(JavaPlugin.class, unused -> {
             // afterEvaluate necessary because the junit-jupiter dep might be added further down the build.gradle
-            project.afterEvaluate(unused -> {
-                project.getConfigurations()
-                        .getByName(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME)
-                        .getAllDependencies()
-                        .matching(dep -> Objects.equals(dep.getGroup(), "org.junit.jupiter")
-                                && dep.getName().equals("junit-jupiter"))
-                        .stream()
-                        .findAny()
-                        .ifPresent(ignored -> enableJUnit5ForAllTestTasks(project));
+            project.afterEvaluate(proj -> {
+                proj.getConvention()
+                        .getPlugin(JavaPluginConvention.class)
+                        .getSourceSets()
+                        .matching(ss -> hasCompileDependenciesMatching(proj, ss, this::isJunitJupiter))
+                        .forEach(ss -> {
+                            log.info("Detected 'org:junit.jupiter:junit-jupiter', enabling useJUnitPlatform()");
+                            String testTaskName = ss.getTaskName(null, "test");
+                            Test testTask = (Test) proj.getTasks().findByName(testTaskName);
+                            if (testTask == null) {
+                                // Fall back to the source set name, since that is what gradle-testsets-plugin does
+                                testTask = (Test) proj.getTasks().findByName(ss.getName());
+                                if (testTask == null) {
+                                    log.warn(
+                                            "Detected 'org:junit.jupiter:junit-jupiter', but unable to find test task");
+                                    return;
+                                }
+                            }
+                            enableJunit5ForTestTask(testTask);
```

Comment:
Wanna move the `log.info` line from 54 to just above here, then we can also include which task name we're enabling it for?